File: 백준/Silver/1018. 체스판 다시 칠하기/체스판 다시 칠하기.cc
#include <iostream>
#include <string>
using namespace std;

string WB[8] = {
	"WBWBWBWB",
	"BWBWBWBW",
	"WBWBWBWB",
	"BWBWBWBW",
	"WBWBWBWB",
	"BWBWBWBW",
	"WBWBWBWB",
	"BWBWBWBW" };

string BW[8] = {
	"BWBWBWBW",
	"WBWBWBWB",
	"BWBWBWBW",
	"WBWBWBWB",
	"BWBWBWBW",
	"WBWBWBWB",
	"BWBWBWBW",
	"WBWBWBWB" };

int cntWB(string* board, int x, int y);
int cntBW(string* board, int x, int y);

int main() {

	int rows;
	int cols;
	cin >> rows >> cols;
	cin.ignore();

	string* board = new string[rows];
	for (int i = 0; i < rows; i++)
	{
		getline(cin, board[i]);
	}


	int minCnt = 64;
	for (int i = 0; i <= rows - 8; i++)
	{
		for (int j = 0; j <= cols - 8; j++)
		{
			if (minCnt >= min(cntWB(board, i, j), cntBW(board, i, j)))
			{
				minCnt = min(cntWB(board, i, j), cntBW(board, i, j));
			}
		}
	}
	cout << minCnt << endl;
}

int cntWB(string* board, int x, int y) {
	int cnt = 0;
	for (int i = 0; i < 8; i++)
	{
		for (int j = 0; j < 8; j++)
		{
			if (board[x + i][y + j] != WB[i][j])
			{
				cnt++;
			}
		}
	}
	return cnt;
}

int cntBW(string* board, int x, int y) {
	int cnt = 0;
	for (int i = 0; i < 8; i++)
	{
		for (int j = 0; j < 8; j++)
		{
			if (board[x + i][y + j] != BW[i][j])
			{
				cnt++;
			}
		}
	}
	return cnt;
}

File: 백준/Silver/1018. 체스판 다시 칠하기/체스판 다시 칠하기.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int rows = sc.nextInt();
        int cols = sc.nextInt();
        sc.nextLine();

        String[] board = new String[rows];
        for (int i = 0; i < rows; i++) {
            board[i] = sc.nextLine();
        }

        int minCnt = 64;
        for (int i = 0; i <= rows - 8; i++) {
            for (int j = 0; j <= cols - 8; j++) {
                int result = Math.min(cntWB(board, i, j), cntBW(board, i, j));
                if (minCnt >= result) {
                    minCnt = result;
                }
            }
        }

        System.out.println(minCnt);
    }

    public static int cntWB(String[] board, int row, int col) {
        String[] WB = {
                "WBWBWBWB",
                "BWBWBWBW",
                "WBWBWBWB",
                "BWBWBWBW",
                "WBWBWBWB",
                "BWBWBWBW",
                "WBWBWBWB",
                "BWBWBWBW"};
        int cnt = 0;

        for (int i = 0; i < 8; i++) {
            for (int j = 0; j < 8; j++) {
                if (board[row + i].charAt(col + j) != WB[i].charAt(j)) {
                    cnt++;
                }
            }
        }
        return cnt;
    }
    public static int cntBW(String[] board, int row, int col) {
        String[] BW = {
                "BWBWBWBW",
                "WBWBWBWB",
                "BWBWBWBW",
                "WBWBWBWB",
                "BWBWBWBW",
                "WBWBWBWB",
                "BWBWBWBW",
                "WBWBWBWB"};
        int cnt = 0;

        for (int i = 0; i < 8; i++) {
            for (int j = 0; j < 8; j++) {
                if (board[row + i].charAt(col + j) != BW[i].charAt(j)) {
                    cnt++;
                }
            }
        }
        return cnt;
    }
}